### PR TITLE
Parse csv

### PR DESCRIPTION
**Pull request by allanchau**
This PR adds a csv parser to the `util.parse` lib code.

### CODE REVIEW

File: idearium-lib/lib/util/parse.js [hunk 1]
```diff
@@ -16,4 +22,52 @@ const parseConfigAsBoolean = (val, defaultVal) => {
...
+
+    let stream = data;
+
+    // Allow the caller to pass a Buffer.
+    if (!(stream instanceof Readable)) {
+
+        stream = new Duplex();
+
+        stream.push(data);
+        stream.push(null);
```
Comment (smebberson) [inline idearium-lib/lib/util/parse.js]: What's the purpose of the `null`. Perhaps you could add a comment?

File: idearium-lib/test/data/test.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+col1,col2,col3,col4,col5
+Test1.1,Test1.2,Test1.3,Test1.4,Test1.5
+Test2.1,Test2.2,Test2.3,Test2.4,Test2.5
+Test3.1,Test3.2,Test3.3,Test3.4,Test3.5
+Test4.1,Test4.2,Test4.3,Test4.4,Test4.5
```
Comment (smebberson) [inline idearium-lib/test/data/test.csv]: Maybe this test data needs to be a little more complex? For example, a row with an empty value in one of the columns would be good. Also, perhaps a column with `""`. Perhaps another column with an emoji? Just to make this more complete and fool proof in future uses.

File: idearium-lib/lib/util/parse.js [hunk 3]
```diff
@@ -16,4 +22,52 @@ const parseConfigAsBoolean = (val, defaultVal) => {
...
+ * @example
+ * parseCsv(BufferData)
+ *     .then(data => console.log(data))
+ *     .catch(console.error);
+ * @param {Buffer|Stream} data Readable stream or buffered data contents.
+ * @param {Object} options Csv-parse options. http://csv.adaltas.com/parse/
+ * @param {Object} options.transform Optional transform function.
+ * @return {Promise} The parsed array contents.
+ */
+const parseCsv = (data, options) => new Promise((resolve, reject) => {
```
Comment (smebberson) [inline idearium-lib/lib/util/parse.js]: Is there a capability to state the header does/doesn't have a header row and either parse or discard the first row accordingly?

File: idearium-lib/test/data/test.csv (hunk 2 above)
Comment (smebberson) [inline idearium-lib/test/data/test.csv]: There should be specific tests for the above suggestions too.

File: idearium-lib/lib/util/parse.js (hunk 3 above)
Comment (allanchau) [inline idearium-lib/lib/util/parse.js]: @smebberson I thought about it, but this parser is really just a dumb parser. All it should do it parse the csv file into a format we can use (array).

Comment (allanchau) [inline idearium-lib/lib/util/parse.js]: header rows can then be handled via `Array.prototype.shift()`

Comment (smebberson) [inline idearium-lib/lib/util/parse.js]: Okay, I agree with this.

**Pull request merged**